src/main/java/blusunrize/immersiveengineering/common/blocks/metal/DieselGeneratorBlockEntity.java

```diff
@@ -172,27 +172,13 @@ public void tickServer()
 			}
 			if(consumeTick<=0) //Consume 10*tick-amount every 10ticks to allow for 1/10th mB amounts
 			{
-				GeneratorFuel recipe = recipeGetter.apply(level, tanks[0].getFluid().getFluid());
-				if(recipe!=null)
+				if (!active)
 				{
-					int burnTime = recipe.getBurnTime();
-					int fluidConsumed = (10*FluidAttributes.BUCKET_VOLUME)/burnTime;
-					if(tanks[0].getFluidAmount() >= fluidConsumed)
```

Comment:
When the amount of fluid in the tank is less than `fuelConsumed`, but is not empty (thus passing the check on line 162), the remaining fuel is not drained. This allows the generator to output flux (line 169) indefinitely.